### ISSUE TITLE
SSL interface missing from successful build of master with -DHAVE_SSL=ON?

### ISSUE THREAD

**Issue by falloutphil** [quickfix/quickfix#360]
Hi,

Build with the following on Ubuntu:
cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo -DHAVE_SSL=ON -DCMAKE_CXX_STANDARD=17

If I use commitid fb2e3cf8f9cdef6e66c546d590797e903052d76a (from July 2019) everything builds as expected and Ican see the SSL interface in the library:

```
$ nm libquickfix.so | grep ThreadedSSLSocketInitiator

0000000000045550 t _GLOBAL__sub_I_ThreadedSSLSocketInitiator.cpp
0000000000103dd0 T _ZN3FIX26ThreadedSSLSocketInitiator11onConfigureERKNS_15SessionSettingsE
0000000000104730 T _ZN3FIX26ThreadedSSLSocketInitiator12onInitializeERKNS_15SessionSettingsE
00000000001041a0 T _ZN3FIX26ThreadedSSLSocketInitiator12removeThreadESt4pairIiP6ssl_stE
0000000000104a50 T _ZN3FIX26ThreadedSSLSocketInitiator12socketThreadEPv
0000000000104330 T _ZN3FIX26ThreadedSSLSocketInitiator16passwordHandleCBEPciiPv
0000000000104300 T _ZN3FIX26ThreadedSSLSocketInitiator22passwordHandleCallbackEPcmiPv
0000000000103c80 T _ZN3FIX26ThreadedSSLSocketInitiator6onPollEd
0000000000105220 T _ZN3FIX26ThreadedSSLSocketInitiator6onStopEv
0000000000105690 T _ZN3FIX26ThreadedSSLSocketInitiator7getHostERKNS_9SessionIDERKNS_10DictionaryERNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERs
0000000000104050 T _ZN3FIX26ThreadedSSLSocketInitiator7onStartEv
0000000000105480 T _ZN3FIX26ThreadedSSLSocketInitiator9addThreadESt4pairIiP6ssl_stEm
0000000000106ee0 T _ZN3FIX26ThreadedSSLSocketInitiator9doConnectERKNS_9SessionIDERKNS_10DictionaryE
0000000000104350 T _ZN3FIX26ThreadedSSLSocketInitiatorC1ERNS_11ApplicationERNS_19MessageStoreFactoryERKNS_15SessionSettingsE
00000000001044e0 T _ZN3FIX26ThreadedSSLSocketInitiatorC1ERNS_11ApplicationERNS_19MessageStoreFactoryERKNS_15SessionSettingsERNS_10LogFactoryE
0000000000104350 T _ZN3FIX26ThreadedSSLSocketInitiatorC2ERNS_11ApplicationERNS_19MessageStoreFactoryERKNS_15SessionSettingsE
00000000001044e0 T _ZN3FIX26ThreadedSSLSocketInitiatorC2ERNS_11ApplicationERNS_19MessageStoreFactoryERKNS_15SessionSettingsERNS_10LogFactoryE
0000000000104710 T _ZN3FIX26ThreadedSSLSocketInitiatorD0Ev
0000000000104670 T _ZN3FIX26ThreadedSSLSocketInitiatorD1Ev
0000000000104670 T _ZN3FIX26ThreadedSSLSocketInitiatorD2Ev
00000000003385b8 V _ZTIN3FIX26ThreadedSSLSocketInitiatorE
0000000000112b40 V _ZTSN3FIX26ThreadedSSLSocketInitiatorE
00000000003385d0 V _ZTVN3FIX26ThreadedSSLSocketInitiatorE
```

Moving forward to master (741e96dbc89988b76ac63e8b036ffac8dcf2bf5c) the exact same command on the same server:

```
$ cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo -DHAVE_SSL=ON -DCMAKE_CXX_STANDARD=17
-- The CXX compiler identification is GNU 7.5.0
-- The C compiler identification is GNU 7.5.0
<snip>
-- Found OpenSSL: /usr/lib/x86_64-linux-gnu/libcrypto.so (found version "1.1.1")
-- OPENSSL_INCLUDE_DIR: /usr/include
-- OPENSSL_LIBRARIES: /usr/lib/x86_64-linux-gnu/libssl.so;/usr/lib/x86_64-linux-gnu/libcrypto.so
-- OPENSSL_ROOT_DIR:
-- Building with SSL
```

However running the same nm command yields no SSL interface, despite:

```
[ 74%] Building CXX object src/C++/CMakeFiles/quickfix.dir/SSLSocketAcceptor.cpp.o
[ 75%] Building CXX object src/C++/CMakeFiles/quickfix.dir/SSLSocketConnection.cpp.o
[ 76%] Building CXX object src/C++/CMakeFiles/quickfix.dir/SSLSocketInitiator.cpp.o
[ 77%] Building CXX object src/C++/CMakeFiles/quickfix.dir/ThreadedSSLSocketAcceptor.cpp.o
[ 77%] Building CXX object src/C++/CMakeFiles/quickfix.dir/ThreadedSSLSocketConnection.cpp.o
[ 78%] Building CXX object src/C++/CMakeFiles/quickfix.dir/ThreadedSSLSocketInitiator.cpp.o
[ 79%] Building CXX object src/C++/CMakeFiles/quickfix.dir/UtilitySSL.cpp.o
[ 80%] Linking CXX shared library libquickfix.so
```

Checking:

```
$ nm libquickfix.so | grep ThreadedSSLSocketInitiator

$ 
```

I have repeated this on 2 different machines using 2 different OSs.

Has the cmake command changed or is there something else I now must do to get SSL to build?

**Comment by KeXu001** [comment]
https://github.com/quickfix/quickfix/pull/350 worked for me

**Comment by leleftheriades** [comment]
be sure to also use the fixes from #274

**Comment by orenmnero** [comment]
Works on the latest master
